Update docs and add  for rust frontend (#13)
nix-community/todomvc-nix: docs/Nix.md

As you can see below, we have `inputs` ([read more](../flake.nix)):
```
{
  description = "todomvc-nix";
  # To update all inputs:
  # $ nix flake update --recreate-lock-file
  inputs.nixpkgs.url = "github:NixOS/nixpkgs/master";
  inputs.flake-utils = {
      url = "github:numtide/flake-utils";
      inputs = {
        nixpkgs.follows = "nixpkgs";
      };
  };
  ...
}
```

which is like every derivation that you want to have in your flakes' project. The syntax is `inputs.<your-chosen-name>.<attribute>`. You can choose whatever name in `<your-chosen-name>` and for the `<attribute>` you have at least:
   1. `url`: for targeting the third-party source code,
   2. `inputs`: if you want to refer to other derivation in `flake.nix`, and
   3. `flake`: Whether you third-party source code have `flake.nix` or not. You can set it to `false` if `flake.nix` is not available.
--------------------------
### **| Flake's outputs attribute |**

`Outputs` attribute contains your project's entry point. It can be used by other project that use flakes feature, or it will be used when you run `nix build`, `nix shell`, or `nix develop` commands.

Here is the snippet of the `outputs` attribute:

> ```

Reviewer: ```suggestion
```nix
```

> outputs = { self, nixpkgs, naersk, ... }: --- (1)
>     {
>       overlay = import ./overlay.nix { inherit  polysemy http-media servant miso servant-jsaddle; }; --- (2)
>     }
>     //
>     (
>       flake-utils.lib.eachSystem [ "x86_64-linux" ] (system: --- (3)
>         let
>           pkgs = import nixpkgs { --- (4)
>             inherit system;
>             config = { ... }; --- (5)
>             overlays = [ ]; --- (6)
>           };
>         in
>         {
>           legacyPackages = pkgs.todomvc; --- (7)
>           packages = flake-utils.lib.flattenTree pkgs.todomvc; --- (8)
>           devShell = import ./devshell.nix { inherit pkgs; }; --- (9)
>           checks = { }; --- (10)
>         }
>       )
>     );
> ```

Reviewer: ```suggestion
outputs = { self, nixpkgs, naersk, ... }: #--- (1)
    {
      overlay = import ./overlay.nix { inherit  polysemy http-media servant miso servant-jsaddle; }; #--- (2)
    }
    //
    (
      flake-utils.lib.eachSystem [ "x86_64-linux" ] (system: #--- (3)
        let
          pkgs = import nixpkgs { #--- (4)
            inherit system;
            config = { ... }; #--- (5)
            overlays = [ ]; #--- (6)
          };
        in
        {
          legacyPackages = pkgs.todomvc; --- (7)
          packages = flake-utils.lib.flattenTree pkgs.todomvc; #--- (8)
          devShell = import ./devshell.nix { inherit pkgs; }; #--- (9)
          checks = { }; #--- (10)
        }
      )
    );
```
```

**(2)** The `overlay` attribute makes every package that we listed in `overlay.nix` available for our `nixpkgs` environment by putting it in `overlays` attribute in line 6.

**(3)** We use `flake-utils` to build a collection of pure Nix functions that don't depend on nixpkgs. The better way to understand it is if you refer to Tweag's blog Part 1, you have to use `defaultPackage.x86_64-linux` in your `outputs` result. With `flake-utils` it has function that make your life more simple by just adding the system in the `flake-utils`'s `eachSystem` argument.

**(4)** We import `nixpkgs` environment. The `nixpkgs` that we use is based on line 1 above.

**(5)** Similar to common `nixpkgs` derivation, you can put system's `config` here (e.g. `allowBroken = true;`).

**(6)** If you refer to this project's overlay, you can see that we put every third-party's overlay in here, such as `devshell.overlay`, `naersk.overlay`, and `mozilla-overlay`. Note that we use `import` on `mozilla-overlay` because flakes feature is not available yet.

**(7)** We put our project's umbrella here, in this case `todomvc` attribute we have in `overlay.nix`. It is used for an o

> **(8)** We use `flattentree` because flakes feature insists on having a flat attribute set of derivations in various places like the `packages` and `checks` attributes. Similar to `legacyPackages`, the `packages` attribute also refer to our project's umbrella.

Reviewer: ```suggestion
**(8)** We use `flattenTree` because flakes feature insists on having a flat attribute set of derivations in various places like the `packages` and `checks` attributes. Similar to `legacyPackages`, the `packages` attribute also refer to our project's umbrella.
```

# TodoMVC-Nix Nix Language

> This section explained all the details about Nix and how it uses in this project.

Reviewer: ```suggestion
This section explains all the details about Nix and how it is being used in this project.
```

> 

Reviewer: What else should be added to this document?

Author: everything, it's just I got stuck with the rust frontend, and just create PR. Will finished it today
